backend/app/interfaces/email_service.py

```diff
@@ -1,4 +1,38 @@
+import json
 from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from enum import Enum
+from typing import Generic, TypeVar
+
+T = TypeVar("T")
+
+
+class TemplateData(ABC):
```

Comment:
Would you be able to move this to a schema file? Given that `routes` is using this too, maybe we should treat this the same as a schema.